libraries/AP_RangeFinder/AP_RangeFinder_TeraRangerI2C.cpp

```diff
@@ -110,44 +110,65 @@ bool AP_RangeFinder_TeraRangerI2C::measure()
     return dev->transfer(&cmd, 1, nullptr, 0);
 }
 
-// collect - return last value measured by sensor
-bool AP_RangeFinder_TeraRangerI2C::collect(uint16_t &_distance_cm)
+// collect_raw() - return last value measured by sensor
+bool AP_RangeFinder_TeraRangerI2C::collect_raw(uint16_t &raw_distance)
 {
     uint8_t d[3];
 
-    // take range reading and read back results
+    // Take range reading
     if (!dev->transfer(nullptr, 0, d, sizeof(d))) {
         return false;
     }
 
+    // Check for CRC
     if (d[2] != crc_crc8(d, 2)) {
-        // bad CRC
         return false;
+    } else {
+         raw_distance = ((uint16_t(d[0]) << 8) | d[1]);
+         return true;
     }
-    
-    _distance_cm = ((uint16_t(d[0]) << 8) | d[1]) / 10;
+}
 
-    return true;
+// Checks for error code and if correct converts to cm
+bool AP_RangeFinder_TeraRangerI2C::check_measure(uint16_t raw_distance, uint16_t &output_distance_cm)
+{
+  // Check for error codes
+  if (raw_distance == 0xFFFF) {
+      // Too far away
+      return false;
+  }
+  if (raw_distance == 0x0000) {
```

Comment:
Can we make those else statements instead?

Reply:
I think it would be better for clarity. To be confirmed.